docs: add FAQ page (#244)
wasmCloud/wasmCloud.com: docs/reference/faq.mdx

### Is wasmCloud Open Source?

Yes! wasmCloud is a [CNCF Sandbox project](https://www.cncf.io/projects/wasmcloud/). wasmCloud has applied for [Incubating status](https://github.com/cncf/toc/pull/1198).

The source code for wasmCloud is available on [GitHub](https://github.com/wasmCloud), under [Version 2 of the Apache License](https://github.com/wasmCloud/wasmCloud/blob/main/LICENSE).

### Is wasmCloud a WebAssembly runtime?

Not exactly. We sometimes refer to wasmCloud [hosts](/docs/concepts/hosts) as "application runtimes." Hosts do run [WebAssembly components](/docs/concepts/webassembly-components), through the use of [wasmtime](https://wasmtime.dev).

### What is the unit of compute for wasmCloud?

wasmCloud applications are composed of [actors](/docs/concepts/actors) and [capability providers](/docs/concepts/capabilities), which are [linked](/docs/concepts/runtime-linking) together at runtime. Decoupling business logic from capabilities is part of the design philosophy of wasmCloud.

### I see wasmCloud actors are stateless, but I need to store state. Is wasmCloud right for me?

> wasmCloud applications can access and store state via [capability providers](/docs/concepts/capabilities). Only actors are stateless.

Reviewer: Do we want to mention in-memory here? Technically actors can store ephemeral state there right?

Reviewer 2: If you mean like accessing a global variable, I _think_ we discard all in-memory information after each invocation ends so nothing would carry over.

Author: Right, actor invocations get fresh state on every invocation. This is by design, so wasmCloud doesn't need to know where to route a particular request; any instance is viable (though NATS leafs will favor local subscribers)

### Do capability providers implement capabilities, or do they act as proxies to external capabilities?

> That depends on the specific capability provider. For example, the [HTTP Server provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/http-server) _is_ a server, while the [Redis provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/kv-redis) acts as a client to an external Redis server.

Reviewer: ```suggestion
Capability providers can both implement capabilities directly and act as proxies to external capabilities, depending on the needs of your application.

For example, the [HTTP Server provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/http-server) _is_ a server, while the [KeyValue ("KV") Redis provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/kv-redis) acts as a client to an external Redis server.
```

### How does wasmCloud handle secrets?

> There are several kinds of secrets involved in distributed applications. Configuration for actors and providers can contain sensitive data. Today, these values are associated with [runtime links](/docs/concepts/runtime-linking), or given to providers [on startup](/docs/cli/ctl#options-3). The host itself is also configured with sensitive data, which are passed to the host [on startup](/docs/reference/host-config).

Reviewer 2: Pending a secrets management RFC, I think it's worth calling out application level secrets.
```suggestion
There are several kinds of secrets involved in distributed applications. Configuration for actors and providers can contain sensitive data. Today, these values are associated with [runtime links](/docs/concepts/runtime-linking), or given to providers [on startup](/docs/cli/ctl#options-3). The host itself is also configured with sensitive data, which are passed to the host [on startup](/docs/reference/host-config).

If you're working with secrets in your application, it's recommended to use a capability provider like [kv-vault](https://github.com/wasmCloud/capability-providers/tree/main/kv-vault) that proxies to a secrets management solution.
```

> ### What is NATS? Why is wasmCloud tightly coupled to NATS?

Reviewer 2: ```suggestion
### What is NATS? Why does wasmCloud depend on NATS?
```

This just feels better to me but feel free to discard

> [NATS](https://nats.io) is open-source "message oriented middleware", responsible for "[addressing, discovery and exchanging of messages that drive the common patterns in distributed systems](https://docs.nats.io/nats-concepts/overview)."

Reviewer 3: Would it be beneficial to spell out what NATS stand for?

Author: Today I learned NATS actually stands for something! https://docs.nats.io/reference/faq#what-does-the-nats-acronym-stand-for 

I don't think wasmCloud users need to know this though
